src/dev/i18n/README.md

```diff
@@ -0,0 +1,208 @@
+# I18n Build Tools
+
+## Default messages extraction tool
+
+### Description
+
+The tool is used to extract default messages from all `*.{js, ts, jsx, tsx, html, handlebars, hbs, jade}` files in provided plugins directories to `translations/en.json` (one JSON per plugin). `translations` directory is created in each of provided plugins directories.\
+The tool uses Babel to parse code and build an AST for each file or JS expression if whole file parsing is impossible. So only static parsing is available for messages extraction. It means that no variables, function calls or expressions can be used for ids, messages and context, only string literals.
+
+### I18n examples and restrictions of the syntax
+
+**Global restriction**: Values of `id`, `defaultMessages` and `context` properties must be string literals. Identifiers, template literals and any other expressions are disallowed.
+
+#### Angular (.html)
+
+* **Filter**
+
+  ```
+  {{ 'plugin_namespace.message_id' | i18n: {
+    values: { /*object*/ },
+    defaultMessage: 'Default message string literal',
+    context: 'Message context'
+  } }}
+  ```
+
+  `values` and `context` properties are optional.\
+  Don't break `| i18n: {` with line breaks, and don't skip whitespaces around `i18n:`.
+
+* **Directive**
+
+  ```html
+  <ANY
+    i18n="plugin_namespace.message_id"
```

Comment:
Actually we have names guideline and examples of using `i18n` plugin in appropriate `README` files. Maybe, let's create link to them for not copypasting information.
And here add main points only regarding what exactly tool does.

Reply:
@yankouskia We need a place for information about syntax restrictions. I think this readme is an appropriate place for it, because syntax is restricted by the extraction tool, not `i18n` plugin.